src/prommis/nanofiltration/diafiltration.py

```diff
@@ -0,0 +1,774 @@
+#####################################################################################################
+# “PrOMMiS” was produced under the DOE Process Optimization and Modeling for Minerals Sustainability
+# (“PrOMMiS”) initiative, and is copyright (c) 2023-2024 by the software owners: The Regents of the
+# University of California, through Lawrence Berkeley National Laboratory, et al. All rights reserved.
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and license information.
+#####################################################################################################
+"""
+Diafiltration cascade flowsheet for separating lithium and cobalt. Serves as an example of a
+custom cost model.
+"""
+
+from pyomo.environ import (
+    ConcreteModel,
+    Expression,
+    Objective,
+    Param,
+    Set,
+    Suffix,
+    TransformationFactory,
+    Var,
+    log,
+    units,
+    value,
+)
+from pyomo.network import Arc
+
+import idaes.logger as idaeslog
+from idaes.core import (
+    FlowsheetBlock,
+    MaterialBalanceType,
+    UnitModelBlock,
+    UnitModelCostingBlock,
+)
+from idaes.core.solvers import get_solver
+from idaes.core.util.initialization import propagate_state
+from idaes.core.util.model_diagnostics import DiagnosticsToolbox
+from idaes.core.util.model_statistics import degrees_of_freedom
+from idaes.models.unit_models import (
+    Mixer,
+    MixerInitializer,
+    MixingType,
+    MomentumMixingType,
+    MSContactor,
+    MSContactorInitializer,
+)
+
+from prommis.nanofiltration.costing.diafiltration_cost_model import (
+    DiafiltrationCosting,
+    DiafiltrationCostingData,
+)
+from prommis.nanofiltration.diafiltration_properties import LiCoParameters
+
+_log = idaeslog.getLogger(__name__)
+
+
+def main():
+    """
+    Builds and solves the diafiltration flowsheet with cost
+
+    Returns:
+        m: Pyomo model
+    """
+    m = build_model()
```

Comment:
This would be `def main(m)` and `build_model(m)`

Reply:
I do not think `main` should take any arguments, as that defeats the purpose of a `main()` method.

Reply:
@andrewlee94 This makes sense and I removed the import of `main` in `test_diafiltration.py`. Now the Codecov check is failing due to uncovered lines in `main()` - I am not sure how to cover those lines if `main()` is no longer being tested.

Reply:
Given this is a test for the case study, you should import and test `main` here. The point of the previous discussion is that `main` should not be viewed as something that other code should important use (and thus it does not need to take an existing model as an argument).

Reply:
Noted, thanks! I'll update this.